runtime/lib/backend/common/Utils.hpp

```diff
@@ -262,16 +264,19 @@ constexpr auto has_gate(const SimulatorGateInfoDataT<size> &arr, const std::stri
     return false;
 }
 
-static inline auto simulateDraw(const std::vector<double> &probs, int8_t postselect) -> bool
+static inline auto simulateDraw(const std::vector<double> &probs, std::optional<int8_t> postselect)
+    -> bool
 {
-    // Check validity of postselect value
-    RT_FAIL_IF(std::abs(postselect) > 1, "Invalid postselect value");
+    if (postselect) {
+        // Check validity of postselect value
+        RT_FAIL_IF(std::abs(*postselect) > 1, "Invalid postselect value");
 
-    // Return the postselect value, do not draw
-    if (postselect != -1) {
-        RT_FAIL_IF(probs[postselect] == 0, "Probability of postselect value is 0");
+        // Return the postselect value, do not draw
+        if (*postselect != -1) {
```

Comment:
C++ code should no longer have to deal with -1 values with the optional type, I think that check  has been replaced by `if (postselect)`.

Reply:
If we do that, some runtime tests fail

Reply:
Done!